### PR TITLE
bingx: add apis

### PR DESCRIPTION
**Pull request by sc0Vu**

### CODE REVIEW

File: ts/src/bingx.ts [hunk 1]
```diff
@@ -100,11 +100,13 @@ export default class bingx extends Exchange {
                                 'trade/openOrders': 3,
                                 'trade/historyOrders': 3,
                                 'account/balance': 3,
+                                'ticker/24hr': 1,
```
Comment (carlosmiei) [inline ts/src/bingx.ts:103]: @sc0Vu maybe we can implement some methods with these endpoints? (fetchTicker spot and cancelOrders spot). Will check it

File: ts/src/bingx.ts [hunk 2]
```diff
@@ -1901,14 +1909,17 @@ export default class bingx extends Exchange {
...
-            throw new BadRequest (this.id + ' cancelOrders is only supported for swap markets.');
-        }
         const request = {
             'symbol': market['id'],
-            'ids': ids,
         };
-        const response = await this.swapV2PrivateDeleteTradeBatchOrders (this.extend (request, params));
+        let response = undefined;
+        if (market['spot']) {
+            request['orderIds'] = ids;
```
Comment (carlosmiei) [inline ts/src/bingx.ts:1917]: hum the signature fails in this endpoint
```
BadRequest bingx {"code":100400,"msg":"Key: 'OrdersCancelRequest.OrderIDs' Error:Field validation for 'OrderIDs' failed on the 'required' tag","debugMsg":""}
```

Comment (sc0Vu) [reply ts/src/bingx.ts:1917]: It looks like the orderIds parameter is not so clear, we might need to check with them. I try both query string and post body, it didn't work.

```
$ node examples/js/cli bingx cancelOrders '["xxx","xxx", "xxx"]' ETH/USDT --verbose
2023-08-24T02:53:31.103Z
Node.js: v20.4.0
CCXT v4.0.72
bingx.cancelOrders (xxx, xxx, xxx, ETH/USDT)
fetch Request:
 bingx POST https://open-api.bingx.com/openApi/spot/v1/trade/cancelOrders?orderIds[]=xxx&orderIds[]=xxx&orderIds[]=xxx&symbol=ETH-USDT&timestamp=xxx&signature=xxx 
RequestHeaders:
 {
  'X-BX-APIKEY': 'xxx',
  'X-SOURCE-KEY': 'CCXT',
} 
RequestBody:
 

handleRestResponse:
 bingx POST bingx POST https://open-api.bingx.com/openApi/spot/v1/trade/cancelOrders?orderIds[]=xxx&orderIds[]=xxx&orderIds[]=xxx&symbol=ETH-USDT&timestamp=xxx&signature=xxx 200 OK 
ResponseHeaders:
 {
  ...
} 
ResponseBody:
 {"code":100400,"msg":"Key: 'OrdersCancelRequest.OrderIDs' Error:Field validation for 'OrderIDs' failed on the 'required' tag","debugMsg":""} 
```

```
$ node examples/js/cli bingx cancelOrders '["xxx","xxx", "xxx"]' ETH/USDT --verbose
2023-08-24T02:53:31.103Z
Node.js: v20.4.0
CCXT v4.0.72
bingx.cancelOrders (xxx, xxx, xxx, ETH/USDT)
fetch Request:
 bingx POST https://open-api.bingx.com/openApi/spot/v1/trade/cancelOrders? 
RequestHeaders:
 {
  'X-BX-APIKEY': 'xxx',
  'X-SOURCE-KEY': 'CCXT',
  'Content-Type': 'application/x-www-form-urlencoded'
} 
RequestBody:
 orderIds[0]=xxx&orderIds[1]=xxx&orderIds[2]=xxx&symbol=ETH-USDT&timestamp=xxx&signature=xxx

handleRestResponse:
 bingx POST https://open-api.bingx.com/openApi/spot/v1/trade/cancelOrders? 200 OK 
ResponseHeaders:
 {
  ...
} 
ResponseBody:
 {"code":100400,"msg":"Key: 'OrdersCancelRequest.OrderIDs' Error:Field validation for 'OrderIDs' failed on the 'required' tag","debugMsg":""} 

```

**Comment by carlosmiei** [comment]
DEMO

```
 bingx createOrder "LTC/USDT" limit buy 0.1 50
Python v3.10.9
CCXT v4.0.72
bingx.createOrder(LTC/USDT,limit,buy,0.1,50)
{'amount': 0.1,
 'average': None,
 'clientOrderId': None,
 'cost': 0.0,
 'datetime': '2023-08-25T10:47:59.870Z',
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': 0.0,
 'id': '1695025223016906752',
 'info': {'cummulativeQuoteQty': '0',
          'executedQty': '0',
          'orderId': '1695025223016906752',
          'origQty': '0.1',
          'price': '50',
          'side': 'BUY',
          'status': 'PENDING',
          'symbol': 'LTC-USDT',
          'transactTime': '1692960479870',
          'type': 'LIMIT'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 0.1,
 'side': 'buy',
 'status': 'open',
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': 1692960479870,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```
```
p bingx fetchOpenOrders "LTC/USDT"             
Python v3.10.9
CCXT v4.0.72
bingx.fetchOpenOrders(LTC/USDT)
[{'amount': 0.1,
  'average': None,
  'clientOrderId': None,
  'cost': 0.0,
  'datetime': '2023-08-25T10:47:59.870Z',
  'fee': {'cost': None, 'currency': None, 'rate': None},
  'fees': [{'cost': None, 'currency': None, 'rate': None}],
  'filled': 0.0,
  'id': '1695025223016906752',
  'info': {'cummulativeQuoteQty': '0',
           'executedQty': '0',
           'orderId': '1695025223016906752',
           'origQty': '0.1',
           'origQuoteOrderQty': '0',
           'price': '50',
           'side': 'BUY',
           'status': 'PENDING',
           'symbol': 'LTC-USDT',
           'time': '1692960479870',
           'type': 'LIMIT',
           'updateTime': '1692960479870'},
  'lastTradeTimestamp': 1692960479870,
  'lastUpdateTimestamp': None,
  'postOnly': None,
  'price': 50.0,
  'reduceOnly': None,
  'remaining': 0.1,
  'side': 'buy',
  'status': 'open',
  'stopLossPrice': None,
  'stopPrice': None,
  'symbol': 'LTC/USDT',
  'takeProfitPrice': None,
  'timeInForce': None,
  'timestamp': 1692960479870,
  'trades': [],
  'triggerPrice': None,
  'type': 'limit'}]
```
```
 p bingx cancelOrders '["1695025223016906752"]' "LTC/USDT"          
Python v3.10.9
CCXT v4.0.72
bingx.cancelOrders(['1695025223016906752'],LTC/USDT)
{'code': '0',
 'data': {'orders': [{'cummulativeQuoteQty': '0',
                      'executedQty': '0',
                      'orderId': '1695025223016906752',
                      'origQty': '0.1',
                      'price': '50',
                      'side': 'BUY',
                      'status': 'CANCELED',
                      'symbol': 'LTC-USDT',
                      'transactTime': '1692960479870',
                      'type': 'LIMIT'}]},
 'debugMsg': '',
 'msg': ''}
```
```
p bingx createOrder "LTC/USDT:USDT" limit buy 1 50       
Python v3.10.9
CCXT v4.0.72
bingx.createOrder(LTC/USDT:USDT,limit,buy,1,50)
{'amount': None,
 'average': None,
 'clientOrderId': None,
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None, 'rate': None},
 'fees': [{'cost': None, 'currency': None, 'rate': None}],
 'filled': None,
 'id': '1695025594552774656',
 'info': {'clientOrderID': '',
          'orderId': '1695025594552774656',
          'positionSide': 'LONG',
          'side': 'BUY',
          'symbol': 'LTC-USDT',
          'type': 'LIMIT'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': 'buy',
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT:USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```
```
p bingx cancelOrders '["1695025594552774656"]' "LTC/USDT:USDT"
Python v3.10.9
CCXT v4.0.72
bingx.cancelOrders(['1695025594552774656'],LTC/USDT:USDT)
{'code': '0',
 'data': {'failed': None,
          'success': [{'advanceAttr': '0',
                       'avgPrice': '0.00',
                       'clientOrderId': '',
                       'commission': '0.000000',
                       'cumQuote': '0',
                       'executedQty': '0.0',
                       'leverage': '',
                       'orderId': '1695025594552774656',
                       'orderType': '',
                       'origQty': '1.0',
                       'positionID': '0',
                       'positionSide': 'LONG',
                       'price': '50.00',
                       'profit': '0.0000',
                       'side': 'BUY',
                       'status': 'CANCELLED',
                       'stopLoss': '',
                       'stopLossEntrustPrice': '0',
                       'stopPrice': '',
                       'symbol': 'LTC-USDT',
                       'takeProfit': '',
                       'takeProfitEntrustPrice': '0',
                       'time': '1692960568451',
                       'type': 'LIMIT',
                       'updateTime': '1692960568466'}]},
 'msg': ''}
```
```
p bingx fetchBalance --swap                                   
Python v3.10.9
CCXT v4.0.72
bingx.fetchBalance()
{'USDT': {'free': 39.8108, 'total': 39.8108, 'used': 0.0},
 'free': {'USDT': 39.8108},
 'info': {'code': '0',
          'data': {'balance': {'asset': 'USDT',
                               'availableMargin': '39.8108',
                               'balance': '39.8108',
                               'equity': '39.8108',
                               'freezedMargin': '0.0000',
                               'realisedProfit': '0.0000',
                               'unrealizedProfit': '0.0000',
                               'usedMargin': '0.0000',
                               'userId': '1177064765068660742'}},
          'msg': ''},
 'total': {'USDT': 39.8108},
 'used': {'USDT': 0.0}}
```
```
p bingx cancelOrder "1695026034740785152" "LTC/USDT:USDT"
Python v3.10.9
CCXT v4.0.72
bingx.cancelOrder(1695026034740785152,LTC/USDT:USDT)
{'amount': 1.0,
 'average': None,
 'clientOrderId': None,
 'cost': 0.0,
 'datetime': '2023-08-25T10:51:13.400Z',
 'fee': {'cost': None, 'currency': None, 'rate': 0.0},
 'fees': [{'cost': None, 'currency': None, 'rate': 0.0}],
 'filled': 0.0,
 'id': '1695026034740785152',
 'info': {'advanceAttr': '0',
          'avgPrice': '0.00',
          'clientOrderId': '',
          'commission': '0.000000',
          'cumQuote': '0',
          'executedQty': '0.0',
          'leverage': '',
          'orderId': '1695026034740785152',
          'orderType': '',
          'origQty': '1.0',
          'positionID': '0',
          'positionSide': 'LONG',
          'price': '50.00',
          'profit': '0.0000',
          'side': 'BUY',
          'status': 'CANCELLED',
          'stopLoss': '',
          'stopLossEntrustPrice': '0',
          'stopPrice': '',
          'symbol': 'LTC-USDT',
          'takeProfit': '',
          'takeProfitEntrustPrice': '0',
          'time': '1692960673400',
          'type': 'LIMIT',
          'updateTime': '1692960673416'},
 'lastTradeTimestamp': 1692960673416,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': 50.0,
 'reduceOnly': None,
 'remaining': 1.0,
 'side': 'buy',
 'status': 'canceled',
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USDT:USDT',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': 1692960673400,
 'trades': [],
 'triggerPrice': None,
 'type': 'limit'}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**